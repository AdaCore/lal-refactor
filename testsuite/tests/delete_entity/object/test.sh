cp ./main.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 2 -R 4
echo "--  (main.orig) -L 2 -R 4"
cat ./main.adb

cp ./main.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 2 -R 7
echo "--  (main.orig) -L 2 -R 7"
cat ./main.adb

cp ./main2.orig ./main.adb
delete_entity -P default.gpr -S main.adb -L 4 -R 7
echo "--  (main2.orig) -L 4 -R 7"
cat ./main.adb
