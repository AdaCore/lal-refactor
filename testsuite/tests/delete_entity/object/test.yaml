title: delete_object

description: delete_enity tool test (object declaration)

driver: shell_script
