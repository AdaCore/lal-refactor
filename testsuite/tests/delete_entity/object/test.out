--  (main.orig) -L 2 -R 4
procedure Main is
   Y : Integer;
begin
   null;
end;
--  main:4:9-4:10 Usage of "Y" that is not safe to delete.
--  (main.orig) -L 2 -R 7
procedure Main is
   X : Integer;
begin
   X := Y + 1;
end;
--  main:10:29-10:30 Usage of "X" that is not safe to delete.
--  (main2.orig) -L 4 -R 7
with Ada.Integer_Text_IO;

procedure Main is
   Y : Integer;

   procedure Get (Value : out Integer) is null;

begin

   Ada.Integer_Text_IO.Put (X);

end;
